src/components/RefinementList/RefinementList.js

```diff
@@ -160,13 +160,19 @@ class RefinementList extends React.Component {
         /> :
         undefined;
 
-    const searchInput = this.props.searchFacetValues ?
-      <SearchBox ref={i => { this.searchbox = i; }}
-        placeholder={this.props.searchPlaceholder}
-        onChange={this.props.searchFacetValues}
-        onValidate={() => this.refineFirstValue()}
-        disabled={!this.props.isFromSearch && displayedFacetValues.length < limit}/> :
-      null;
+    let searchInput = null;
+    if (this.props.searchFacetValues) {
+      searchInput = this.props.searchIsAlwaysActive ?
+        <SearchBox ref={i => { this.searchbox = i; }}
+          placeholder={this.props.searchPlaceholder}
+          onChange={this.props.searchFacetValues}
+          onValidate={() => this.refineFirstValue()}/> :
+        <SearchBox ref={i => { this.searchbox = i; }}
+          placeholder={this.props.searchPlaceholder}
+          onChange={this.props.searchFacetValues}
+          onValidate={() => this.refineFirstValue()}
+          disabled={!this.props.isFromSearch && displayedFacetValues.length < limit}/>;
```

Comment:
Why not change this `this.props.isFromSearch` to be a const we decide somewhere else, based on `searchIsAlwaysActive` as well?